spec/classes/foreman_plugin_discovery_spec.rb

```diff
@@ -7,12 +7,18 @@
 
     let(:facts) { facts }
```

Comment:
Got it!  This should be inside the `context "on #{os}"` block.

When you run the whole file you're getting facts from the wrong OS, probably whichever OS is defined first or last.  This might be why it was affecting Ruby 1.8.7 tests in particular as the hash order isn't deterministic - while on other Ruby versions it was probably a single OS every time (something with /var/lib).